manager/main.go

```diff
@@ -135,16 +141,27 @@ func main() {
 	}
 }
 
-func setupReconcilers(ctx context.Context, setupLog logr.Logger, mgr ctrl.Manager) {
+type closable interface {
+	Close(ctx context.Context) error
+}
+
+type noOpCloser struct{}
+
+func (c noOpCloser) Close(ctx context.Context) error {
+	return nil
+}
+
+func setupReconcilers(ctx context.Context, setupLog logr.Logger, mgr ctrl.Manager) closable {
 	if features.IsActive(features.FullLifecycleAPI()) {
-		setupFullLifecycleReconcilers(ctx, setupLog, mgr)
-	} else {
-		setupLog.Info("Setting up legacy cluster controller")
-		setupLegacyClusterReconciler(setupLog, mgr)
+		return setupFullLifecycleReconcilers(ctx, setupLog, mgr)
 	}
+
+	setupLog.Info("Setting up legacy cluster controller")
+	setupLegacyClusterReconciler(setupLog, mgr)
+	return noOpCloser{}
```

Comment:
just to clarify, we're returning the no-op here because the legacy controller already has logic to handle closing the dependencies?

Reply:
Because the legacy controller doesn't have any dependency to close (afaik)